Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -164,17 +174,16 @@ internal void OnException(Exception exception)
         /// </summary>
         internal void OnActivityStart(HttpRequestMessage request)
         {
-            if (Activity.Current == null)
+            var currentActivity = Activity.Current;
+            if (currentActivity == null)
             {
                 DependencyCollectorEventSource.Log.CurrentActivityIsNull();
                 return;
             }
 
-            if (request != null &&
-                !this.applicationInsightsUrlFilter.IsApplicationInsightsUrl(request.RequestUri))
```

Comment:
This chack is [moved to](https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/993cab25f0076ef6942ad6e1a640e18fcc14fac1/Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs#L451) IsEnabled callback

So AI internal calls are rejected a bit earlier, which allows to save a bit on HttpClient instrumentation and  DiagnosticSource payloads and  Activity objects creation.

At this stage of request processing in .netcore  (when the event is fired from HttpClient) request could not be null, HttpClient will throw before that.